src/ensembl/src/content/app/entity-viewer/gene-view/components/protein-domain-image/ProteinDomainImage.tsx

```diff
@@ -0,0 +1,178 @@
+
+  return (
+    <div className={styles.container}>
+      {/* TODO: The sorting needs to be done based on the `score`? once it is available */}
+      {Object.keys(proteinDomainsResources)
+        .sort()
+        .map((type, key) => {
+          return (
+            <div key={key} className={styles.resourceGroup}>
+              <div className={styles.resourceName}>{type}</div>
+              <div className={styles.resourceImages}>
+                {Object.keys(proteinDomainsResources[type])
+                  .sort()
+                  .map((description, key) => {
+                    return (
+                      <div key={key} className={styles.resourceImage}>
+                        <svg
+                          className={styles.containerSvg}
+                          width={props.width}
+                          height={BLOCK_HEIGHT}
+                        >
+                          <g className={domainClasses}>
+                            <Backbone {...props} scale={scale} />
+                            {proteinDomainsResources[type][description].map(
+                              (exon, index) => (
+                                <ExonBlock
+                                  key={index}
+                                  exon={exon}
+                                  transcriptStart={transcriptStart}
+                                  className={props.classNames?.exon}
+                                  scale={scale}
+                                />
+                              )
+                            )}
+                          </g>
+                        </svg>
+                        <div className={styles.resourceDescription}>
+                          {description}
+                        </div>
+                      </div>
+                    );
+                  })}
+              </div>
+            </div>
+          );
+        })}
+    </div>
+  );
+};
+
+const Backbone = (
+  props: ProteinDomainImageProps & { scale: ScaleLinear<number, number> }
+) => {
+  const backboneClasses = classNames(
+    styles.backbone,
+    props.classNames?.backbone
+  );
+
+  // we are adjusting the width of every backbone segment by adding 2 points from its width.
+  return (
+    <g className={backboneClasses}>
+      <rect height={BACKBONE_HEIGHT} width={props.width + 2} />
+    </g>
+  );
+};
+
+type ExonBlockProps = {
+  exon: {
+    start: number;
+    end: number;
+  };
+
+  transcriptStart: number;
+  className?: string;
+  scale: ScaleLinear<number, number>;
+};
+
+const ExonBlock = (props: ExonBlockProps) => {
+  const y = 3;
+  const exonClasses = classNames(styles.exon, props.className);
+
+  const classes = classNames(exonClasses);
+
+  return (
+    <rect
+      key={props.exon.start}
+      className={classes}
+      y={y}
+      height={BLOCK_HEIGHT}
+      x={props.scale(props.exon.start)}
+      width={props.scale(props.exon.end - props.exon.start)}
```

Comment:
end minus start plus one, I am afraid 😃 I think I forgot to do that for transcripts.

Reply:
Why do we need to do plus one here?

Reply:
Because we need to count the nucleotide that is not taken into account. Check [this comment](https://github.com/Ensembl/ensembl-client/pull/273#discussion_r406176713) for the explanation.

Reply:
> Why do we need to do plus one here?

Humans start counting at 1 and stop counting at the last position of the counted objects.

For example, for a sequence of two nucleotides, the position of the first nucleotide is 1, the position of the last one is 2, and the length of the sequence is end position - start position + 1 (i.e. 2 - 1 + 1, which is 2).

Reply:
Thanks @imransl & @azangru. I've updated it.